alg/teca_temporal_reduction.cxx

```diff
@@ -0,0 +1,648 @@
+#include "teca_temporal_reduction.h"
+
+#include "teca_cartesian_mesh.h"
+#include "teca_array_collection.h"
+#include "teca_variant_array.h"
+#include "teca_metadata.h"
+#include "teca_calendar_util.h"
+
+#include <algorithm>
+#include <iostream>
+#include <string>
+#include <cmath>
+
+#if defined(TECA_HAS_BOOST)
+#include <boost/program_options.hpp>
+#endif
+
+#define TECA_DEBUG
+
+using std::string;
+using std::vector;
+using std::cerr;
+using std::endl;
+using std::cos;
+
+// --------------------------------------------------------------------------
+void summation_operator::initialize(double fill_value)
+{
+    this->fill_value = fill_value;
+}
+
+// --------------------------------------------------------------------------
+void summation_operator::update(
+    p_teca_variant_array &out_array,
+    p_teca_variant_array &out_valid,
+    const_p_teca_variant_array &in_array,
+    const_p_teca_variant_array &in_valid)
+{
+    for (unsigned int i = 0; i < out_array->size(); ++i)
```

Comment:
pull the call to `size` outside of the loop's condition. Function's in the condition get called before each iteration of the loop. There is more of a chance for compiler optimization when the condition is a simple comparison.